File: macos/BolsaTerminal/Core/Models/Quote.swift
import Foundation

struct QuoteSummary: Codable, Hashable, Identifiable {
    var id: String { symbol }
    let symbol: String
    let shortName: String
    let longName: String?
    let exchange: String
    let currency: String
    let regularMarketPrice: Double
    let regularMarketChange: Double
    let regularMarketChangePercent: Double
    let regularMarketVolume: Double
    let regularMarketOpen: Double?
    let regularMarketDayHigh: Double?
    let regularMarketDayLow: Double?
    let regularMarketPreviousClose: Double?
    let fiftyTwoWeekHigh: Double?
    let fiftyTwoWeekLow: Double?
    let marketCap: Double?
    let trailingPE: Double?
    let forwardPE: Double?
    let dividendYield: Double?
    let beta: Double?
    let averageVolume: Double?
    let sector: String?
    let industry: String?
    let quoteType: String
    let marketState: String
}

File: android/core/model/src/main/java/com/bolsaterminal/core/model/Quote.kt
package com.bolsaterminal.core.model

import kotlinx.serialization.Serializable

@Serializable
data class QuoteSummary(
    val symbol: String,
    val shortName: String,
    val longName: String? = null,
    val exchange: String,
    val currency: String,
    val regularMarketPrice: Double,
    val regularMarketChange: Double,
    val regularMarketChangePercent: Double,
    val regularMarketVolume: Double,
    val regularMarketOpen: Double? = null,
    val regularMarketDayHigh: Double? = null,
    val regularMarketDayLow: Double? = null,
    val regularMarketPreviousClose: Double? = null,
    val fiftyTwoWeekHigh: Double? = null,
    val fiftyTwoWeekLow: Double? = null,
    val marketCap: Double? = null,
    val trailingPE: Double? = null,
    val forwardPE: Double? = null,
    val dividendYield: Double? = null,
    val beta: Double? = null,
    val averageVolume: Double? = null,
    val sector: String? = null,
    val industry: String? = null,
    val quoteType: String,
    val marketState: String,
)
